Automattic/pocket-casts-android
# [Project] Manage Downloaded Episodes - Display Modal when running on low storage and turn on Auto Download

## Description

- Displays the low storage modal when running on low storage and **turn on Auto Download**
- This PR does not implement the snooze for 7 days feature when taping on Maybe Later button
- See: [VvERCFIopLjFL8Fnczsf8r-fi-123_1167](https://href.li/?https://www.figma.com/file/VvERCFIopLjFL8Fnczsf8r/?node-id=123-1167)
- Context: p1730298732804909/1730291928.354409-slack-C07TL9YK9V1

Fixes #3150

## Testing Instructions
Modify the code bellow to always return `true` to make the tests faster. This will mock the low storage check
https://github.com/Automattic/pocket-casts-android/blob/565d4f7ef92be14a27bc7cfe3c6aa6a24962a178/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/StorageUtil.kt#L11

1. Download an episode
2. Go to Profile -> Auto Download
3. Turn on New Episodes toggle
4. ✅ Ensure the modal opens 

## Screenshots or Screencast 
[Screen_recording_20241101_133738.webm](https://github.com/user-attachments/assets/0283e083-2254-464e-9038-67e20508dfdb)


## Checklist
 
#### I have tested any UI changes...
